From 128ea7cb0f238e5157e2c438dfc5c21ce995d7fb Mon Sep 17 00:00:00 2001
From: "tong.gao" <tong.gao@amlogic.com>
Date: Fri, 14 Jul 2023 13:45:57 +0800
Subject: [PATCH 3/3] t7b: rx enc [1/1]

PD#SWPL-131661

Problem:
can not work

Solution:
add tvin-port

Verify:
local

Signed-off-by: tong.gao <tong.gao@amlogic.com>
Change-Id: I116bae23f42bdddbfcc33236a3ee96c2a1c5e652
---
 amlv4l2src/amlv4l2src.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/amlv4l2src/amlv4l2src.c b/amlv4l2src/amlv4l2src.c
index ce00499..6b25057 100644
--- a/amlv4l2src/amlv4l2src.c
+++ b/amlv4l2src/amlv4l2src.c
@@ -166,18 +166,21 @@ int get_default_tvin_port(const char** devname){
   }
 
   // HDMI RX
-  if (0 == strcmp("vdinvideo",(char*)(cap.driver))) {
+  if (0 == strcmp("vdinvideo",(char*)(cap.driver)) || 0 == strcmp("vdin_video",(char*)(cap.driver))) {
     // VPP0(Video)
+    printf("hdmirx tvin-port use 0 \n");
     return 0;
   }
   // screen
   if (0 == strcmp("amlvideo2",(char*)(cap.driver))) {
     // VPP0(OSD+Video)
+    printf("screen tvin-port use 0x11000001 \n");
     return 0x11000001;
   }
 
   // VPP0(OSD+Video)
-  return 0x11000001;
+  printf("error tvin-port use -1 \n");
+  return -1;
 }
 
 static int get_correspond_media_node(const char *video_filepath, char *media_filepath){
-- 
2.25.1

